File: customer-java-resttemplate/src/main/java/com/prez/lib/health/info/OtherInfosContributor.java
package com.prez.lib.health.info;

import java.time.OffsetDateTime;
import org.springframework.boot.actuate.info.Info;
import org.springframework.boot.actuate.info.InfoContributor;

public class OtherInfosContributor implements InfoContributor {

    private static final OffsetDateTime START_DATE = OffsetDateTime.now();

    private boolean isSnapshot;
    private String applicationType;

    public OtherInfosContributor(boolean isSnapshot, String applicationType) {
        this.isSnapshot = isSnapshot;
        this.applicationType = applicationType;
    }

    @Override
    public void contribute(Info.Builder builder) {
        builder.withDetail("snapshot", isSnapshot);
        builder.withDetail("startDate", START_DATE.toString());
        builder.withDetail("applicationType", applicationType);
    }

}

File: customer-kotlin-webflux/src/main/kotlin/com/prez/lib/health/info/OtherInfosContributor.kt
package com.prez.lib.health.info

import org.springframework.boot.actuate.info.Info
import org.springframework.boot.actuate.info.InfoContributor
import java.time.OffsetDateTime

class OtherInfosContributor(private val isSnapshot: Boolean, private val applicationType: String) : InfoContributor {

  companion object {
    private val START_DATE = OffsetDateTime.now()
  }

  override fun contribute(builder: Info.Builder) {
    builder.withDetail("snapshot", isSnapshot)
    builder.withDetail("startDate", START_DATE.toString())
    builder.withDetail("applicationType", applicationType)
  }
}